src/packages/auth/src/authentication/methods/utils.ts

```diff
@@ -70,14 +77,63 @@ export const runAfterHooks = async <D, H = CredentialInsertInput | CredentialUpd
 	return hookEntities;
 };
 
-export const maskSensitiveValuesForLogging = (
+const obfuscateSensitiveValues = (
+	variables: VariableValues | undefined,
+	nodeNamesToMask: Set<string>
+) => {
+	try {
+		if (!variables) return variables;
+
+		const obfuscateNode = (key: string, value: any): any => {
+			if (nodeNamesToMask.has(key)) {
+				return '********';
+			}
+
+			if (value === null || value === undefined) return value;
+
+			if (Array.isArray(value)) {
+				return value.map((item: any) => obfuscateNode(key, item));
+			}
+
+			if (typeof value === 'symbol') return '********'; // technically impossible, but just in case
+
+			if (typeof value === 'object') {
+				for (const newKey in value) {
+					value[newKey] = obfuscateNode(newKey, value[newKey]);
+				}
+			}
+
+			return value;
+		};
+
+		const variablesCopy = JSON.parse(JSON.stringify(variables));
+
+		for (const key in variablesCopy) {
```

Comment:
I prefer to combine the clone and then value iteration using Object.fromEntries, like in this example:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/fromEntries#object_transformations